File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.model.Cart;

import java.util.List;

/**
 *  @author: guanx
 *  @Date: 2020/3/3 13:40
 *  @Description: 购物车接口
 */
public interface CartService {

    /**
    * @Description 加入购物车
    * @Author  guanx
    * @Date   2020/3/3 13:40
    * @Param
    * @Return
    * @Exception
    *
    */
    List<Cart> add(List<Cart> cartList,Long itemId, Integer num);

    /**
     * @Description 加入购物车 到Redis
     * @Author  guanx
     * @Date   2020/3/3 13:40
     * @Param
     * @Return
     * @Exception
     *
     */
    void addGoodsToRedis(String userName,List<Cart> cartList);
    
    /**
    * @Description 查询购物车 从Redis
    * @Author  guanx
    * @Date   2020/3/3 20:09
    * @Param  
    * @Return      
    * @Exception   
    * 
    */
    List<Cart> findCartListFromRedis(String userName);

    /**
     * @Description 合并cookie信息和redis信息
     * @Author  guanx
     * @Date   2020/3/3 20:57
     * @Param
     * @Return
     * @Exception
     *
     */
    List<Cart> mergeList(List<Cart> redisCartList,List<Cart> cookieCartList);
}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
/**
* @Description 购物车Service
* @Author  guanx
* @Date   2020/3/3 15:33
* @Param  
* @Return      
* @Exception   
* 
*/
app.service('cartService',function ($http) {

    //查询购物车数据
    this.findCartList=function(){
        return $http.post('/cart/cartList.shtml');
    }

    //添加购物车
    this.addCart=function (itemId,num) {
        return $http.get('/cart/add.shtml?itemId=' + itemId + '&num=' + num);
    }

    //查询用户地址
    this.getAddressList=function () {
        return $http.get('/address/user/list.shtml');
    }
})